### PR TITLE
Style tweak, animation revamp and more

### LINKED ISSUE #51: Re-trigger voting animation
## Background

No matter we show the result or not, it's better experience if we could know any player is re-voting.

## Normal solutions

To restart animation, we have several approaches.

- Use `setTimeout` to put the next state change into next tick of event loop

```javascript
this.setState({ voted: false }, () => {
  setTimeout(() => this.setState({ voted: true }), 0);
});
```

- Move `voted` out of `<Card />` to destroy (`voted=false`) then re-build the DOM (`voted=true`)

```javascript
{voted && <Card voted={voted} />}
```

- Trigger re-flow or listen on `webkitAnimationEnd` event. https://css-tricks.com/restart-css-animation/

## Issues in poker

As for our poker, there are two kinds of cards.

- Me

We could easily pass down a prop `me={true}` to `<Card />` for indicating it's me, and check `nextProps.score !== this.props.score` in `componentWillReceiveProps` hook to indicate if I re-vote, then invoke one of the approaches above to restart animation.

- Other players

To avoid knowing others' score before we show the result, poker server will return `null` for other players' score, that means we could not use `nextProps.score !== this.props.score` to indicate if other player re-vote. Without this if statement, other players' card will always re-animate even he/she doesn't re-vote but other actions trigger the whole state update.

---

Open this ticket here to discuss.

### PR DESCRIPTION
**Pull request by hyjk2000**
This resolves #51.

Dependency: [poker-server#13](https://github.com/IFS49F/poker-server/pull/13)

- The card bounces when player votes, and now when re-votes as well
- Player joining & leaving are now animated and more conspicuous (no more sneaking in or out)
- Corresponding votes are now highlighted when mouseover on summary scores
- Card suit ♤♧♡♢ is now randomized for each player 
- Minor cosmetic changes to cards for a lighter feeling

### CODE REVIEW

**Comment by just4fun** [comment]
tenor-222636649

Pretty cool! Let me try it in the weekend!

**Comment by just4fun** [comment]
CR and DEV-QA done. Just have some questions above.

**Comment by just4fun** [comment]
I really like the animation, but I'm not good at CSS 😢 

Why are you so DIAO?

File: src/lib/player-state-pinger.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      const { me, team } = nextState;
+      const player = [me, ...team].find(client => client.id === playerId);
+      player[state] = value;
+      return nextState;
+    });
+  }
+
+  clearTimeouts() {
+    for (let elm in this.timeouts) {
+      if (typeof elm === 'object') this.clearTimeouts(elm);
```
[the remark this answers is not in the document]
Comment (hyjk2000) [reply src/lib/player-state-pinger.js]: Yeah, I'm deep traversing the object recursively. Any better idea?

File: src/pages/Room/Room.js [hunk 2]
```diff
@@ -68,6 +73,26 @@ class Room extends Component {
...
+  handleAction(action) {
+    switch (action.type) {
+      case 'vote':
+        this.playerStatePinger.pingPlayerState(action.playerId, 'voting', 1000);
+        break;
+      default:
+    }
+  }
+
+  setPlayerState(playerId, state, value) {
```
[the remark this answers is not in the document]
Comment (hyjk2000) [reply src/pages/Room/Room.js]: Oops, that's what happens if you make last minute changes.

**Comment by just4fun** [comment]
Ready to go.

**Pull request merged**